File: elixir/run_all.exs
#!/usr/bin/env elixir

# Run all Elixir performance benchmarks

Mix.install([
  {:httpoison, "~> 2.0"},
  {:jason, "~> 1.4"}
])

# Load all benchmark modules
Code.require_file("lib/sorting.ex", __DIR__)
Code.require_file("lib/fibonacci.ex", __DIR__)
Code.require_file("lib/matrix.ex", __DIR__)
Code.require_file("lib/strings.ex", __DIR__)
Code.require_file("lib/api_requests.ex", __DIR__)

IO.puts("\n" <> String.duplicate("=", 60))
IO.puts("Elixir Performance Benchmarks")
IO.puts(String.duplicate("=", 60) <> "\n")

# Run all benchmarks
IO.puts("\n=== 1. Sorting (Quicksort) ===")
Sorting.run_benchmark()

IO.puts("\n=== 2. Fibonacci ===")
Fibonacci.run_benchmark()

IO.puts("\n=== 3. Matrix Multiplication ===")
Matrix.run_benchmark()

IO.puts("\n=== 4. String Manipulation ===")
Strings.run_benchmark()

IO.puts("\n=== 5. API Requests ===")
metrics = ApiRequests.run_benchmark()
ApiRequests.print_results(metrics)

IO.puts("\n" <> String.duplicate("=", 60))
IO.puts("All benchmarks completed!")
IO.puts(String.duplicate("=", 60) <> "\n")

File: javascript/run_all.js
#!/usr/bin/env node
/**
 * Run all JavaScript benchmarks.
 */
const sorting = require('./sorting');
const fibonacci = require('./fibonacci');
const matrix = require('./matrix');
const strings = require('./strings');
const apiRequests = require('./api_requests');

async function main() {
    console.log('='.repeat(60));
    console.log('JavaScript Performance Benchmarks');
    console.log('='.repeat(60));
    console.log();
    
    // Sorting
    console.log('Running Sorting Benchmark...');
    const sortingResult = sorting.runBenchmark();
    console.log(`  ${sortingResult.testName}: ${sortingResult.executionTimeMs.toFixed(2)} ms`);
    console.log();
    
    // Fibonacci
    console.log('Running Fibonacci Benchmarks...');
    const fibResults = fibonacci.runBenchmark();
    fibResults.forEach(result => {
        console.log(`  ${result.testName}: ${result.executionTimeMs.toFixed(2)} ms`);
    });
    console.log();
    
    // Matrix
    console.log('Running Matrix Multiplication Benchmark...');
    const matrixResult = matrix.runBenchmark();
    console.log(`  ${matrixResult.testName}: ${matrixResult.executionTimeMs.toFixed(2)} ms`);
    console.log();
    
    // Strings
    console.log('Running String Manipulation Benchmarks...');
    const stringResults = strings.runBenchmark();
    stringResults.forEach(result => {
        console.log(`  ${result.testName}: ${result.executionTimeMs.toFixed(2)} ms`);
    });
    console.log();
    
    // API Requests
    console.log('Running API Request Benchmark...');
    const apiResult = await apiRequests.runBenchmark();
    console.log(`  ${apiResult.testName}: ${apiResult.executionTimeMs.toFixed(2)} ms`);
    console.log(`  Successful: ${apiResult.successfulRequests}/${apiResult.totalRequests}`);
    console.log();
    
    console.log('='.repeat(60));
    console.log('All benchmarks completed!');
    console.log('='.repeat(60));
}

main().catch(console.error);
